pkg/publicapi/client/v2/client.go

```diff
@@ -80,20 +96,36 @@ func (c *httpClient) write(ctx context.Context, verb, endpoint string, in apimod
 	if r.BodyObj == nil && r.Body == nil {
 		r.BodyObj = in
 	}
-	_, resp, err := requireOK(c.doRequest(ctx, verb, endpoint, r)) //nolint:bodyclose // this is being closed
-	if err != nil && resp != nil && resp.StatusCode == http.StatusUnauthorized {
-		return apimodels.ErrInvalidToken
-	} else if err != nil {
+
+	_, resp, err := c.doRequest(ctx, verb, endpoint, r) //nolint:bodyclose // this is being closed
+	defer resp.Body.Close()
+	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
```

Comment:
Move the defer after the error check. If an error is returned, the response will likely be nil, and attempting to close a nil response will panic.